cpp/tests/stream_compaction/drop_duplicates_tests.cpp

```diff
@@ -14,254 +14,245 @@
+    input, keys, cudf::duplicate_keep_option::KEEP_FIRST, null_equality::EQUAL);
+  CUDF_TEST_EXPECT_TABLES_EQUAL(input, got_sort->view());
 
-  cudf::size_type expected = std::set<T>(input.begin(), input.end()).size();
-  EXPECT_EQ(expected,
-            cudf::distinct_count(input_col, null_policy::INCLUDE, nan_policy::NAN_IS_VALID));
+  auto got_unordered = unordered_drop_duplicates(input, keys, null_equality::EQUAL);
+  CUDF_TEST_EXPECT_TABLES_EQUAL(input, got_unordered->view());
 }
 
-TEST_F(DistinctCount, WithNansOnly)
+TEST_F(DropDuplicateCommon, NoColumnInputTable)
 {
-  using T = float;
+  cudf::table_view input{std::vector<cudf::column_view>()};
+  std::vector<cudf::size_type> keys{1, 2};
 
-  std::vector<T> input               = {1, 3, NAN, 70, 31};
-  std::vector<cudf::size_type> valid = {1, 1, 1, 1, 1};
+  auto got =
+    drop_duplicates(input, keys, cudf::duplicate_keep_option::KEEP_FIRST, null_equality::EQUAL);
+  CUDF_TEST_EXPECT_TABLES_EQUAL(input, got->view());
 
-  cudf::test::fixed_width_column_wrapper<T> input_col{input.begin(), input.end(), valid.begin()};
+  auto got_sort = sort_and_drop_duplicates(
+    input, keys, cudf::duplicate_keep_option::KEEP_FIRST, null_equality::EQUAL);
+  CUDF_TEST_EXPECT_TABLES_EQUAL(input, got_sort->view());
 
-  cudf::size_type expected = 5;
-  EXPECT_EQ(expected,
-            cudf::distinct_count(input_col, null_policy::INCLUDE, nan_policy::NAN_IS_VALID));
+  auto got_unordered = unordered_drop_duplicates(input, keys, null_equality::EQUAL);
+  CUDF_TEST_EXPECT_TABLES_EQUAL(input, got_unordered->view());
 }
 
-TEST_F(DistinctCount, NansAsNullWithNoNull)
+TEST_F(DropDuplicateCommon, EmptyKeys)
 {
-  using T = float;
+  cudf::test::fixed_width_column_wrapper<int32_t> col{{5, 4, 3, 5, 8, 1}, {1, 0, 1, 1, 1, 1}};
+  cudf::test::fixed_width_column_wrapper<int32_t> empty_col{};
+  cudf::table_view input{{col}};
+  std::vector<cudf::size_type> keys{};
 
-  std::vector<T> input               = {1, 3, NAN, 70, 31};
-  std::vector<cudf::size_type> valid = {1, 1, 1, 1, 1};
+  auto got =
+    drop_duplicates(input, keys, cudf::duplicate_keep_option::KEEP_FIRST, null_equality::EQUAL);
+  CUDF_TEST_EXPECT_TABLES_EQUAL(cudf::table_view{{empty_col}}, got->view());
 
-  cudf::test::fixed_width_column_wrapper<T> input_col{input.begin(), input.end(), valid.begin()};
+  auto got_sort = sort_and_drop_duplicates(
+    input, keys, cudf::duplicate_keep_option::KEEP_FIRST, null_equality::EQUAL);
+  CUDF_TEST_EXPECT_TABLES_EQUAL(cudf::table_view{{empty_col}}, got_sort->view());
 
-  cudf::size_type expected = 5;
-  EXPECT_EQ(expected,
-            cudf::distinct_count(input_col, null_policy::INCLUDE, nan_policy::NAN_IS_NULL));
+  auto got_unordered = unordered_drop_duplicates(input, keys, null_equality::EQUAL);
+  CUDF_TEST_EXPECT_TABLES_EQUAL(cudf::table_view{{empty_col}}, got_unordered->view());
 }
 
-TEST_F(DistinctCount, NansAsNullWithNull)
+struct DropDuplicate : public cudf::test::BaseFixture {
```

Comment:
This should be named like the function it calls:
```suggestion
struct DropDuplicates : public cudf::test::BaseFixture {
```